talk/owt/sdk/base/peerconnectiondependencyfactory.cc

```diff
@@ -126,6 +128,13 @@ void PeerConnectionDependencyFactory::
     if (!GlobalConfiguration::GetCustomizedVideoDecoderEnabled()) {
       decoder_factory.reset(new MSDKVideoDecoderFactory());
     }
+  } else {
+    if (!encoded_frame_) {
+      encoder_factory = std::move(webrtc::CreateBuiltinVideoEncoderFactory());
+    }
+    if (!GlobalConfiguration::GetCustomizedVideoDecoderEnabled()) {
```

Comment:
this adds handling when hardware_acceleration is not true. not related with line 142.

Reply:
Refactored the code here to be more reasonable.